lib/ansible/modules/windows/win_shortcut.ps1

```diff
@@ -64,75 +80,75 @@ If ($state -eq "absent") {
-            $result.changed = $true
-            $ShortCut.Arguments = $args
+        If (($null -ne $arguments) -and ($ShortCut.Arguments -ne $arguments)) {
+            $module.Result.changed = $true
+            $ShortCut.Arguments = $arguments
         }
-        $result.args = $ShortCut.Arguments
+        $module.Result.args = $ShortCut.Arguments
 
-        If ($directory -ne $null -and $ShortCut.WorkingDirectory -ne $directory) {
-            $result.changed = $true
+        If (($null -ne $directory) -and ($ShortCut.WorkingDirectory -ne $directory)) {
+            $module.Result.changed = $true
             $ShortCut.WorkingDirectory = $directory
         }
-        $result.directory = $ShortCut.WorkingDirectory
+        $module.Result.directory = $ShortCut.WorkingDirectory
 
         # FIXME: Not all values are accepted here ! Improve docs too.
-        If ($hotkey -ne $null -and $ShortCut.Hotkey -ne $hotkey) {
-            $result.changed = $true
+        If (($null -ne $hotkey) -and ($ShortCut.Hotkey -ne $hotkey)) {
+            $module.Result.changed = $true
             $ShortCut.Hotkey = $hotkey
         }
-        $result.hotkey = $ShortCut.Hotkey
+        $module.Result.hotkey = $ShortCut.Hotkey
 
-        If ($icon -ne $null -and $ShortCut.IconLocation -ne $icon) {
-            $result.changed = $true
+        If (($null -ne $icon) -and ($ShortCut.IconLocation -ne $icon)) {
+            $module.Result.changed = $true
             $ShortCut.IconLocation = $icon
         }
-        $result.icon = $ShortCut.IconLocation
+        $module.Result.icon = $ShortCut.IconLocation
 
-        If ($orig_description -ne $null -and $ShortCut.Description -ne $description) {
-            $result.changed = $true
+        If (($null -ne $description) -and ($ShortCut.Description -ne $description)) {
+            $module.Result.changed = $true
             $ShortCut.Description = $description
         }
-        $result.description = $ShortCut.Description
+        $module.Result.description = $ShortCut.Description
 
-        If ($windowstyle -ne $null -and $ShortCut.WindowStyle -ne $windowstyles.$windowstyle) {
-            $result.changed = $true
+        If (($null -ne $windowstyle) -and ($ShortCut.WindowStyle -ne $windowstyles.$windowstyle)) {
+            $module.Result.changed = $true
             $ShortCut.WindowStyle = $windowstyles.$windowstyle
         }
-        $result.windowstyle = $windowstyleids[$ShortCut.WindowStyle]
+        $module.Result.windowstyle = $windowstyleids[$ShortCut.WindowStyle]
     }
 
-    If ($result.changed -eq $true -and $check_mode -ne $true) {
+    If (($module.Result.changed -eq $true) -and ($module.CheckMode -ne $true)) {
         Try {
             $ShortCut.Save()
         } Catch {
-            Fail-Json -obj $result -message "Failed to create shortcut '$dest'. ($($_.Exception.Message))"
+            $module.FailJson("Failed to create shortcut '$dest'. ($($_.Exception.Message))")
```

Comment:
Add `$_` to the 2nd argument for FailJson to include a better error with `-vvv`.

Reply:
Done.